src/library/traktor/traktorfeature.cpp

```diff
@@ -337,6 +419,28 @@ void TraktorFeature::parseTrack(QXmlStreamReader &xml, QSqlQuery &query) {
                 bpm = attr.value("BPM").toString().toFloat();
                 continue;
             }
+            if (xml.name() == "CUE_V2") {
+                QXmlStreamAttributes attr = xml.attributes ();
+                int hotcue = attr.value("HOTCUE").toString().toInt();
+                //int position = int(attr.value("START").toString().toFloat()/1000.0 * float(samplerate) * 2.0);
+                int position = int(attr.value("START").toString().toDouble()/1000.0 * 48000.0 * 2.0);
```

Comment:
We need here to use the tack sample rate.

We are moving to a floating point transport engine. This means Mixxx can handle double precision cue points soon, and the code here should use double to not loose info.

Reply:
The Track sample rate is not available at this point.
It is assumed to be 48kHz at first, but the moment the sample rate is available (on track import [traktorfeature.cpp:59 and 79]), the position is changed to the proper value.

Did I misunderstand the second part of the comment? 
Because this code uses double already.

Reply:
I mean: int position -> double position 
but maybe I have missed something,

Reply:
Oh ok.
Is the position not denoting the sample?
What exactly would a double value mean in this case?

Reply:
Ah ... posMultiplier. Can't we just remove the 48000, in all places and user positionMs until it is converted into Mixxx stereo samples playposition?

Reply:
Yeah it's probably a prettier solution not to do it the current way.
My main motivation was to keep the traktor_cue table in the same scheme as the cue table.
If changing the position in the traktor_cue table to double, we could insert the raw value of Traktor and later convert it.

What do you mean with "and user positionMs until it is converted into Mixxx stereo samples playposition"?

Reply:
Exactly what you wrote before. The fancy stereo sample position is called "playposition" in the Mixxx code base. The traktor position needs an other name,for example positionMs. 

> What exactly would a double value mean in this case?

If you play a track at 50 % speed for instance, you are able to set a cue point between two samples.

Reply:
I changed the way the traktor cues are saved.
Now I keep the native format for the database.
The conversion is done later and in a single position.
This makes the position conversion more easy and future proof.
(Now you only have to change a single line if the playposition is changed to double)